File: dart/lib/src/models/stats_breakdown.dart
//
//  rerout
//  stats_breakdown.dart
//
//  Official Dart SDK for the Rerout branded-link API.
//
//  Created by Ngonidzashe Mangudya on 20/05/2026.
//  Copyright (c) 2026 Codecraft Solutions
//  Licensed under the MIT License
//  https://codecraftsolutions.co.za
//

import 'package:meta/meta.dart';

/// A single bucket in an analytics breakdown — e.g. one country, one device.
@immutable
class StatsBreakdown {
  /// Creates a [StatsBreakdown].
  const StatsBreakdown({required this.value, required this.clicks});

  /// Parses one bucket from the API JSON envelope.
  factory StatsBreakdown.fromJson(Map<String, dynamic> json) => StatsBreakdown(
    value: json['value'] as String? ?? '',
    clicks: (json['clicks'] as num?)?.toInt() ?? 0,
  );

  /// The bucket label — country code, device class, browser name, etc.
  final String value;

  /// Click count for this bucket.
  final int clicks;
}

File: java/src/main/java/co/rerout/sdk/model/StatsBreakdown.java
/*
 * rerout-java — Official Java SDK for the Rerout branded-link API.
 *
 * Copyright (c) 2026 Codecraft Solutions
 * Licensed under the MIT License — https://codecraftsolutions.co.za
 */

package co.rerout.sdk.model;

/** A single bucket in an analytics breakdown — e.g. one country, one device. */
public final class StatsBreakdown {

    private final String value;
    private final long clicks;

    /**
     * Creates a {@code StatsBreakdown}.
     *
     * @param value  the bucket label — country code, device class, browser name
     * @param clicks the click count for this bucket
     */
    public StatsBreakdown(String value, long clicks) {
        this.value = value;
        this.clicks = clicks;
    }

    /** {@return the bucket label} */
    public String getValue() {
        return value;
    }

    /** {@return the click count for this bucket} */
    public long getClicks() {
        return clicks;
    }
}
